### DISCUSSION TITLE
Logging Gain in local Hubs instance

### DISCUSSION THREAD

**Discussion by rishivinod7** [mozilla/hubs#3940] [General Questions & Discussion] [answered]
Would anyone know how to log the gain value, or more specifically which file I should log the gain in? I'm currently experimenting with changing the avatar distance models and therefore trying to see the change in gain values for different scenarios. At the moment I've added this line of code in the audio-system.js file:
console.log("Gain: "+ this.outboundGainNode.gain.value);

But with no meaningful feedback. Should I be looking to log it in the avatar-volume-controls.js file?

**Comment by netpro2k** [comment, marked as answer]
It sounds like you are wanting to play with the `PannerNode` settings. These wont be reflected at all in the `GainNode`'s gain value (the only thing that will change that are the volume buttons on the avatar).

The relevant (default) settings are here https://github.com/mozilla/hubs/blob/master/src/components/media-views.js#L205-L211 for media and here for avatars https://github.com/mozilla/hubs/blob/master/src/components/avatar-audio-source.js#L49-L55 (though note, avatar's default value is overridden in hub.html and both defaults can be overridden by the scene).

For a description of what these values mean you can check out https://developer.mozilla.org/en-US/docs/Web/API/PannerNode.

I don't suspect you are going to be able to boil it down to a single value since the `PannerNode` is doing more than just distance based volume attenuation, but if you want to try and get something like that you can probably do so with an https://developer.mozilla.org/en-US/docs/Web/API/AnalyserNode (we actually do that to provide audio feedback like head scaling and mouth flaps, see audio-feedback.js)

**Reply by rishivinod7** [reply]
Thanks! This is useful in terms of pointing me to a better source, in terms of using the analyser node to visualise the output in the form of an oscilloscope or frequency graph; would you have any advice or guidance as to how this should be carried out? I'm assuming I would need to bind the data from audio-feedback.js to a component in hubs.js to draw it, but I'm finding it a bit confusing as my understanding of Node/react is not the best.
